Northstar.Client/mod/scripts/vscripts/ui/menu_mode_select.nut

```diff
@@ -1,86 +1,552 @@
+untyped
 global function InitModesMenu
+global function NSSetModeCategory
+
+global enum eModeMenuModeCategory
+{
+	UNKNOWN = 0,
+	PVPVE   = 1,
+	PVE     = 2,
+	PVP     = 3,
+	FFA     = 4,
+	TITAN   = 5,
+	OTHER   = 6,
+	CUSTOM  = 7
+
+	SIZE
+}
+
+// List of blocked modes due to them being unfinished
+const array<string> blockedModes =
+[
+	"at",
+	"fd_easy",
+	"fd_normal",
+	"fd_hard",
+	"fd_master",
+	"fd_insane"
+]
+
+struct ListEntry_t {
+	string mode
+	int category
+}
+
+// Slider mouse delta buffer
+struct {
+	int deltaX = 0
+	int deltaY = 0
+} mouseDeltaBuffer
 
 struct {
-	int currentModePage
+	int scrollOffset
+	var menu
+
+	string searchString
+	int searchEnum
+
+	// Table of category overrides
+	table<string,int> categoryOverrides
+
+	// List of all modes we know
+	array<ListEntry_t> modes
+
+	// Sorted list of modes we want to show with categories included
+	array<string> sortedModes
 } file
 
 const int MODES_PER_PAGE = 15
```

Comment:
Since you're not using pages, why keeping this?

Reply:
The scroll bar logic uses this in multiple places. Imo it's better to keep this as a constant for readability.